parsers/jscript.c

```diff
@@ -2398,10 +2415,47 @@ static bool parsePrototype (tokenInfo *const name, tokenInfo *const token, state
 					skipArgumentList(method_body_token, false,
 									 vStringLength (signature) == 0 ? signature : NULL);
 				else
+				{
+					if ( isType (method_body_token, TOKEN_EQUAL_SIGN) &&
+						! isType (identifier_token, TOKEN_UNDEFINED) &&
+						prototype_token_count < sizeof(prototype_tokens)/sizeof(tokenInfo*) - 1)
+					{
+						tokenInfo *const temp_token =
+							prototype_tokens[ prototype_token_count++ ] = newToken ();
+						copyToken (temp_token, identifier_token, true);
+					}
+					else if ( isType(method_body_token, TOKEN_IDENTIFIER))
+						copyToken (identifier_token, method_body_token, false);
+
 					readToken (method_body_token);
+				}
 			}
 
 			int index = makeJsTag (token, JSTAG_METHOD, signature, NULL);
+
+			tokenInfo ** temp_token = prototype_tokens;
+			while (*temp_token != NULL)
+			{
+				char * s1 = vStringValue((*temp_token)->string);
+				char * s2 = strstr(s1, ".prototype.");
+				if (s2 != NULL)
+				{
+					s1[ s2-s1 ] = 0;	// terminate before "prototype"
+					vStringSetLength((*temp_token)->string);
```

Comment:
See vStringTruncate. It may be what you want.

Reply:
I don't want to lose the text after `.prototype.`